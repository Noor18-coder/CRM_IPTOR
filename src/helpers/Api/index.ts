export * from './ApiRequest';
export * from './User';
export * from './CompanyInfo';
export * from './CurrencyInfo';
export * from './StagesInfo';
export * from './OpportunityType';
export * from './Attributes';
