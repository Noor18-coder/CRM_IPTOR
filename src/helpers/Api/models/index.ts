export * from './UserItem';
export * from './CompanyInfo';
export * from './ApiRequestControls';
export * from './Opportunity';
export * from './OpportunityDetails';
export * from './Users';
export * from './StageInfo';
export * from './Contacts';
export * from './Product';
export * from './Customer';
export * from './OpportunityTypes';
export * from './CustomerDetails';
export * from './AddOpportunity';
export * from './Items';
export * from './OpportunityAttributes';
export * from './Currency';
export * from './Attributes';
export * from './AddCustomer';
export * from './Countries';
export * from './ForeCastInfo';
