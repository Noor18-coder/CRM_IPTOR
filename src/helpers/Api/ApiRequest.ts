/**
 * Api Request Object
 */
import { ApiRequestControls } from './models';

export class ApiRequest<T> {
  IptorAPI: string;

  id: string;

  constructor(public method: string, public params?: T, public control?: ApiRequestControls) {
    this.IptorAPI = process.env.REACT_APP_IPTOR_API_VERSION ?? '';
    this.id = process.env.REACT_APP_IPTOR_API_ID ?? '';
  }
}
