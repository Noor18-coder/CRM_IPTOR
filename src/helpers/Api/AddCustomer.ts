import axios, { AxiosResponse } from 'axios';
import { get } from 'lodash';
import { AddBusinessPartnerDefaultParams, AddBusinessPartnerResponse, SaveUserDefinedFieldParam } from './models';
import { ApiRequest } from './ApiRequest';

export default class AddCustomerApi {
  /** API Method */
  private static apiMethod = 'crmBusinessPartner.add';

  private static updateApiMethod = 'crmBusinessPartner.update';

  private static saveAttributeValueMethod = 'mopAttribute.add';

  private static saveAttributeValueMethodFile = 'SRONAM';

  /**
   * Helper function to fetch Business Partner Info
   * API Method: crmBusinessPartner.add
   * */
  static async add(businessPartner: AddBusinessPartnerDefaultParams): Promise<AddBusinessPartnerResponse> {
    const requestData = new ApiRequest<AddBusinessPartnerDefaultParams>(this.apiMethod, businessPartner);
    const response = await axios.post<AxiosResponse>('/api/service', requestData);
    return get(response, 'data');
  }

  static async update(businessPartner: AddBusinessPartnerDefaultParams): Promise<AddBusinessPartnerResponse> {
    const requestData = new ApiRequest<AddBusinessPartnerDefaultParams>(this.updateApiMethod, businessPartner);
    const response = await axios.post<AxiosResponse>('/api/service', requestData);
    return get(response, 'data');
  }

  static async addAttributes(businessPartnerId: string, attributeType: string, attributeValue: string | number): Promise<AddBusinessPartnerResponse> {
    const params: SaveUserDefinedFieldParam = {
      attributeType,
      parentFile: this.saveAttributeValueMethodFile,
      parentId: businessPartnerId,
      attributeValue,
    };
    const requestData = new ApiRequest<SaveUserDefinedFieldParam>(this.saveAttributeValueMethod, params);
    const response = await axios.post<AxiosResponse>('/api/service', requestData);
    return get(response, 'data');
  }
}
