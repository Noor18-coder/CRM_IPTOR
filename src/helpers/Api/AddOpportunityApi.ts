import axios, { AxiosResponse } from 'axios';
import { AddOpportunityDefaultParams, AddOpportunityResponse, SaveUserDefinedFieldParam, AddItemToOpportunityParams, AddItemToOpportunityResponse, AddCustomerContactParams, AddCustomerContactRequestParams } from './models';
import { ApiRequest } from './ApiRequest';
import { get } from 'lodash';

export default class AddOpportunityApi {
  /** API Method */
  private static apiMethod: string = 'mopOpportunity.add';
  private static saveAttributeValueMethod: string = 'mopAttribute.add';
  private static saveAttributeValueMethodFile: string = 'SROMOPH';
  private static saveItemMethod: string = 'mopItem.add';
  private static saveContactFileName: string = 'SROMOPCH';
  private static saveContactMethod: string = 'mopContact.add';

  /**
   * Helper function to fetch Business Partner Info
   * API Method: mopOpportunity.get
   * @param freeTextSearch search term
   * 
   */
  static async add(opportunity: AddOpportunityDefaultParams): Promise<AddOpportunityResponse> {
    const requestData = new ApiRequest<AddOpportunityDefaultParams>(this.apiMethod, opportunity);
    const response = await axios.post<AxiosResponse>('/api/service', requestData);
    return get(response, 'data');
  }

  /**
   * Helper function to fetch Business Partner Info
   * API Method: mopOpportunity.get
   * @param freeTextSearch search term
   * 
   */
  static async addAttributes(opportunityId: string, attributeType: string, attributeValue: string | number): Promise<AddOpportunityResponse> {
    const params: SaveUserDefinedFieldParam = {
      attributeType: attributeType,
      parentFile: this.saveAttributeValueMethodFile,
      parentId: opportunityId,
      attributeValue: attributeValue
    }
    const requestData = new ApiRequest<SaveUserDefinedFieldParam>(this.saveAttributeValueMethod, params);
    const response = await axios.post<AxiosResponse>('/api/service', requestData);
    return get(response, 'data');
  }

  /**
   * Helper function to fetch Business Partner Info
   * API Method: mopOpportunity.get
   * @param freeTextSearch search term
   * 
   */
  static async addItem(opportunityId: string, item: string, quantity: number, unit:string): Promise<AddItemToOpportunityResponse> {
    const params: AddItemToOpportunityParams = {
      parentFile: this.saveAttributeValueMethodFile,
      parentId: opportunityId,
      item: item,
      quantity: quantity,
      unit:unit
    }
    const requestData = new ApiRequest<AddItemToOpportunityParams>(this.saveItemMethod, params);
    const response = await axios.post<AddItemToOpportunityResponse>('/api/service', requestData);
    return get(response, 'data');
  }

  /**
   * Helper function to fetch Business Partner Info
   * API Method: mopOpportunity.get
   * @param freeTextSearch search term
   * 
   */
   static async addContacts(params:AddCustomerContactParams): Promise<AxiosResponse> {
    const request:AddCustomerContactRequestParams  = {
      contactParentFile:this.saveAttributeValueMethodFile,
      contactParentId:params.contactPArentId,
      contactPerson:params.contactPerson,
      phone:params.phone,
      mobile:params.mobile,
      contactDC:'',
      whatsApp:'',
      linkedin:'',
      fax:'',
      email:params.email
    }

    const requestData = new ApiRequest<AddCustomerContactRequestParams>(this.saveContactMethod, request);
    const response = await axios.post<AxiosResponse>('/api/service', requestData);
    console.log(response);
    return get(response, 'data');
  }
}
