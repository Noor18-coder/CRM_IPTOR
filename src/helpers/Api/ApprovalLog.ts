import axios, { AxiosResponse } from 'axios';
import { get } from 'lodash';
import { ApprovalLogAddRequestParams, ApprovalLogAddResponse, DeleteApprovalLogsParams } from './models';
import { ApiRequest } from './ApiRequest';

export class ApprovalLog {
  /** API Method */
  private static apiMethod = 'crmOpportunityApprovalLog.add';

  private static deleteApprovalLogs = 'crmOpportunityApprovalLog.delete';

  /**
   * Helper function to fetch sales order list
   * @param params user id
   * @returns information about user
   */
  static async addApprovalLog(params: ApprovalLogAddRequestParams): Promise<ApprovalLogAddResponse> {
    const requestData = new ApiRequest<ApprovalLogAddRequestParams>(this.apiMethod, params);
    const response = await axios.post<ApprovalLogAddResponse>('/api/service', requestData);
    return get<AxiosResponse<ApprovalLogAddResponse>, 'data'>(response, 'data');
  }

  static async deleteOpportunityApprovalLogs(params: DeleteApprovalLogsParams): Promise<ApprovalLogAddResponse> {
    const requestData = new ApiRequest<DeleteApprovalLogsParams>(this.deleteApprovalLogs, params);
    const response = await axios.post<ApprovalLogAddResponse>('/api/service', requestData);
    return get(response, 'data.data.items', {});
  }
}
