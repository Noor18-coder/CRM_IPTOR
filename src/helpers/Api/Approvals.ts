import axios, { AxiosResponse } from 'axios';
import { get } from 'lodash';
import * as apiModels from './models';
import { ApiRequest } from './ApiRequest';

export class ApprovalInfo {
  /** API Method */
  private static allApprovalsMethod = 'crmOpportunityApprovalLogs.get';

  private static allNotesMethod = 'mopNotes.get';

  private static notesParentFile = 'SROMOPAL';

  static async getApprovalLogs(opportunityId: string): Promise<apiModels.ApprovalLogsDefault[]> {
    const requestData = new ApiRequest<apiModels.ApprovalLogsParams>(this.allApprovalsMethod, { opportunityId });
    const response = await axios.post<apiModels.ApprovalLogsResponse>('/api/service', requestData);
    return get<AxiosResponse<apiModels.ApprovalLogsResponse>, 'data', 'data', 'items', apiModels.ApprovalLogsDefault[]>(
      response,
      ['data', 'data', 'items'],
      []
    );
  }

  static async getAllNotes(parentId: string): Promise<apiModels.AllNotesDefault> {
    const requestData = new ApiRequest<apiModels.AllNotesParams>(this.allNotesMethod, { parentFile: this.notesParentFile, parentId });
    const response = await axios.post<apiModels.AllNotesResponse>('/api/service', requestData);
    const defaultData = {} as apiModels.AllNotesDefault;
    return get<AxiosResponse<apiModels.AllNotesResponse>, 'data', 'data', 'items', apiModels.AllNotesDefault>(
      response,
      ['data', 'data', 'items'],
      defaultData
    );
  }
}
