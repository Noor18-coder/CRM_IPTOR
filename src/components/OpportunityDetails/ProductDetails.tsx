import React from 'react';
import { Card, Image, Table, Accordion } from 'react-bootstrap';
import { useMediaQuery } from 'react-responsive';
import { Dispatch } from 'redux';
import { useSelector, useDispatch } from 'react-redux';
import i18n from '../../i18n';
import { AttributeValueObject, Product } from '../../helpers/Api/models';
import * as models from '../../helpers/Api/models';
import ImageConfig from '../../config/ImageConfig';
import { AppState } from '../../store/store';
import { getDateInFormat } from '../../helpers/utilities/lib';
import {
  getOpportunityProducts,
  getProductInformation,
  openOpportunityForm,
  deleteOpportunityProducts,
} from '../../store/OpportunityDetails/Actions';

export interface ContactProps {
  data: models.Product;
  opportunityId: string;
}

interface Props {
  opportunityId: string;
}

export const ProductAccordian: React.FC<Props> = ({ opportunityId }) => {
  const isMobile = useMediaQuery({ maxWidth: 767.98 });
  const isTablet = useMediaQuery({ minWidth: 768, maxWidth: 991 });

  const [activeClass, setActiveClass] = React.useState('');
  const state: AppState = useSelector((appState: AppState) => appState);
  const dispatch: Dispatch<any> = useDispatch();
  const { products } = state.opportuntyDetails;
  const toggleAccordion = () => {
    setActiveClass(activeClass === '' ? 'active' : '');
  };
  const openAddItemForm = (str: string) => {
    document.body.classList.add('body-scroll-hidden');
    dispatch(openOpportunityForm({ open: true, groupName: str, action: 'edit' }));
  };

  React.useEffect(() => {
    dispatch(getOpportunityProducts(opportunityId));
  }, []);

  return (
    <Accordion defaultActiveKey="0">
      <Card className="add-details">
        <Accordion.Toggle className={activeClass} onClick={toggleAccordion} as={Card.Link} eventKey="1">
          {i18n.t('productSectionTitle')}
          {state.opportuntyDetails.editOportunity.allowEdit ? (
            <Image className="addnew-red-icon" src={ImageConfig.ADD_BTN} alt="Add New" title="Add New" onClick={() => openAddItemForm('add_item')} />
          ) : null}
        </Accordion.Toggle>
        <Accordion.Collapse eventKey="1">
          <div className="accr-body-container prod-modules-tbl mob-prod-modules-tbl">
            {(() => {
              if (products.length) {
                if (isMobile || isTablet) {
                  return products.map((obj: Product) => {
                    return <ProductCards data={obj} opportunityId={opportunityId} />;
                  });
                } else {
                  return (
                    <Table borderless>
                      <thead>
                        <tr>
                          <th>{i18n.t('itemId')}</th>
                          <th>{i18n.t('itemName')}</th>
                          {state.enviornmentConfigs.productAttributes.slice(0, 5).map((obj: models.AttributeField) => {
                            return <th>{obj.description}</th>;
                          })}
                        </tr>
                      </thead>
                      <tbody>
                        {state.opportuntyDetails.products.length &&
                          state.opportuntyDetails.products.map((obj: Product) => {
                            return <ProductCardsTable data={obj} opportunityId={opportunityId} />;
                          })}
                      </tbody>
                    </Table>
                  );
                }
              } else {
                return <div className="no-prod-data-txt"> No Records Found </div>;
              }
            })()}
          </div>
        </Accordion.Collapse>
      </Card>
    </Accordion>
  );
};

export const ProductCards: React.FC<ContactProps> = ({ data, opportunityId }) => {
  const state: AppState = useSelector((appState: AppState) => appState);
  React.useEffect(() => {
    dispatch(getProductInformation(data.itemId));
  }, []);

  const dispatch: Dispatch<any> = useDispatch();

  const openAddItemForm = (action: string) => {
    if (action === 'delete_item') {
      dispatch(deleteOpportunityProducts(opportunityId, data));
    } else {
      document.body.classList.add('body-scroll-hidden');
      dispatch(openOpportunityForm({ open: true, groupName: action, data, action: 'edit' }));
    }
  };

  const getValue = (obj: models.AttributeField) => {
    const attributes: models.AttributeValueObject[] | undefined =
      data &&
      data.attributes &&
      data.attributes.filter((dataAttributeValue: AttributeValueObject) => {
        return obj.attributeType === dataAttributeValue.attributeType;
      });

    let result = '';

    if (!attributes) {
      return result;
    }

    attributes.forEach((attribute: models.AttributeValueObject) => {
      const value = attribute && attribute.attributeValue ? attribute.attributeValue : '--';
      if (obj?.valueFormatDesc === 'DATE') {
        const year = value.substring(0, 4);
        const month = value.substring(4, 6);
        const day = value.substring(6, 8);
        result += `${year}-${month}-${day}, `;
        const tempDate = `${year}-${month}-${day}`;
        result += `${getDateInFormat(new Date(tempDate))}, `;
      } else if (obj?.valueFormatDesc === 'BOOLEAN') {
        result += `${value ? (value === 'Y' ? 'Yes' : 'No') : '--'}, `;
      } else {
        result += `${value}, `;
      }
    });
    result = result.slice(0, -2);
    if (result === '') {
      return '--';
    } else {
      return result;
    }
  };

  return (
    <>
      <div className="mr-10 mob-prod-modules-card">
        <div className="d-flex justify-content-between product-row">
          <div className="lft-col">
            {i18n.t('itemId')} <span>{data.item}</span>
          </div>
          <div className="rgt-col">
            {i18n.t('itemName')} <span className="danger">{data.itemDescription}</span>
          </div>
        </div>
        <div className="d-flex justify-content-between product-row prod-mod-collection">
          {state.enviornmentConfigs.productAttributes.slice(0, 5).map((obj: models.AttributeField, index: number) => {
            if (index % 2 === 0) {
              return (
                <div className="lft-col">
                  {obj.description} <div className="values">{getValue(obj)}</div>
                </div>
              );
            } else {
              return (
                <div className="rgt-col">
                  {obj.description} <div className="values">{getValue(obj)}</div>
                </div>
              );
            }
          })}
        </div>

        <div className="d-flex justify-content-between product-row action-row">
          <button
            type="button"
            className=" d-flex justify-content-between product-row icon-class mr-6"
            onClick={() => openAddItemForm('delete_item')}>
            <span className="icon">
              <Image className="del-icon" src={ImageConfig.DELETE_ICON} alt="edit" title="edit" />
              Delete
            </span>
          </button>
          <button type="button" className=" d-flex justify-content-between product-row icon-class" onClick={() => openAddItemForm('edit_item')}>
            <span className="icon">
              <Image className="del-icon" src={ImageConfig.EDIT_ICON} alt="edit" title="edit" />
              Edit
            </span>
          </button>
        </div>
      </div>
    </>
  );
};

export const ProductCardsTable: React.FC<ContactProps> = ({ opportunityId, data }) => {
  const state: AppState = useSelector((appState: AppState) => appState);
  // const data: models.Product | undefined = state.opportuntyDetails.products.find((obj: models.Product) => obj.itemId === itemId);
  const dispatch: Dispatch<any> = useDispatch();
  const openAddItemForm = (action: string) => {
    if (action === 'delete_item') {
      dispatch(deleteOpportunityProducts(opportunityId, data));
    } else {
      document.body.classList.add('body-scroll-hidden');
      dispatch(openOpportunityForm({ open: true, groupName: action, data, action: 'edit' }));
    }
  };

  const getValue = (obj: models.AttributeField) => {
    const attributes: models.AttributeValueObject[] | undefined =
      data &&
      data.attributes &&
      data.attributes.filter((dataAttributeValue: AttributeValueObject) => {
        return obj.attributeType === dataAttributeValue.attributeType;
      });

    let result = '';

    if (!attributes) {
      return result;
    }

    attributes.forEach((attribute: models.AttributeValueObject) => {
      const value = attribute && attribute.attributeValue ? attribute.attributeValue : '';
      if (obj?.valueFormatDesc === 'DATE') {
        if (value !== '0') {
          const year = value.substring(0, 4);
          const month = value.substring(4, 6);
          const day = value.substring(6, 8);
          const tempDate = `${year}-${month}-${day}`;
          result += `${getDateInFormat(new Date(tempDate))}, `;
        } else {
          result = '--';
        }
      } else if (obj?.valueFormatDesc === 'BOOLEAN') {
        result += `${value ? (value === 'Y' ? 'Yes' : 'No') : '--'}, `;
      } else {
        result += `${value}, `;
      }
    });
    result = result.slice(0, -2);
    if (result === '') {
      return '--';
    } else {
      return result;
    }
  };

  return (
    <>
      {data ? (
        <tr>
          <td className="prod-class">{data.item}</td>
          <td className="prod-class">{data.itemDescription}</td>
          {state.opportuntyDetails.products
            ? state.enviornmentConfigs.productAttributes.slice(0, 5).map((obj: models.AttributeField) => {
                return <td className="prod-class">{getValue(obj)}</td>;
              })
            : null}
          <td className="prod-revenue-class">
            {state.opportuntyDetails.editOportunity.allowEdit ? (
              <div className="d-flex justify-content-between title-row float-right">
                <button type="button" className="lft-col" onClick={() => openAddItemForm('delete_item')}>
                  <Image src={ImageConfig.CLOSE_BTN} alt="delete" title="delete" />
                </button>
                <button type="button" className="rgt-col" onClick={() => openAddItemForm('edit_item')}>
                  <Image src={ImageConfig.EDIT_ICON} alt="edit" title="edit" />
                </button>
              </div>
            ) : null}
          </td>
        </tr>
      ) : null}
    </>
  );
};
