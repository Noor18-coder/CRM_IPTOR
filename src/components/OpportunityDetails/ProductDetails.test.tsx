import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { ProductAccordian, ProductCards, ProductCardsTable } from './ProductDetails';
import StoreMock from '../../mocks/Store.mock';
import { Product } from '../../helpers/Api/models';

jest.mock('axios');
const store = StoreMock.createInitialStore();
const product = {} as Product;

describe('[OpportunityDetails] ProductAccordian', () => {
  it('should renders correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <ProductAccordian opportunityId="opportunityId" />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('[OpportunityDetails] ProductCards', () => {
  it('should renders correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <ProductCards opportunityId="opportunityId" data={product} />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('[OpportunityDetails] ProductCardsTable', () => {
  it('should renders correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <ProductCardsTable opportunityId="opportunityId" data={product} />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
