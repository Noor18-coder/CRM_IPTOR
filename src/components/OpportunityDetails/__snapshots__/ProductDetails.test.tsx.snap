// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[OpportunityDetails] ProductAccordian should renders correctly 1`] = `
<div
  className="accordion"
>
  <div
    className="add-details card"
  >
    <a
      className="card-link"
      onClick={[Function]}
    >
      Products and Modules
    </a>
    <div
      aria-expanded={null}
      className="collapse"
    >
      <div
        className="accr-body-container prod-modules-tbl mob-prod-modules-tbl"
      >
        <div
          className="no-prod-data-txt"
        >
           No Records Found 
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`[OpportunityDetails] ProductCards should renders correctly 1`] = `
<div
  className="mr-10 mob-prod-modules-card"
>
  <div
    className="d-flex justify-content-between product-row"
  >
    <div
      className="lft-col"
    >
      Item ID
       
      <span />
    </div>
    <div
      className="rgt-col"
    >
      Item Name
       
      <span
        className="danger"
      />
    </div>
  </div>
  <div
    className="d-flex justify-content-between product-row prod-mod-collection"
  />
  <div
    className="d-flex justify-content-between product-row action-row"
  >
    <button
      className=" d-flex justify-content-between product-row icon-class mr-6"
      onClick={[Function]}
      type="button"
    >
      <span
        className="icon"
      >
        <img
          alt="edit"
          className="del-icon"
          src="delete-icon.svg"
          title="edit"
        />
        Delete
      </span>
    </button>
    <button
      className=" d-flex justify-content-between product-row icon-class"
      onClick={[Function]}
      type="button"
    >
      <span
        className="icon"
      >
        <img
          alt="edit"
          className="del-icon"
          src="edit-icon.svg"
          title="edit"
        />
        Edit
      </span>
    </button>
  </div>
</div>
`;

exports[`[OpportunityDetails] ProductCardsTable should renders correctly 1`] = `
<tr>
  <td
    className="prod-class"
  />
  <td
    className="prod-class"
  />
  <td
    className="prod-revenue-class"
  />
</tr>
`;
